dmoj/graders/signature.py

```diff
@@ -24,12 +24,9 @@ def _generate_binary(self):
             entry_point = self.problem.problem_data[handler_data['entry']]
             header = self.problem.problem_data[handler_data['header']]
 
-            # fmt: off
-            submission_prefix = (
-                '#include "%s"\n'
-                '#define main main_%s\n'
-            ) % (handler_data['header'], str(uuid.uuid4()).replace('-', ''))
-            # fmt: on
+            submission_prefix = '#include "%s"\n' % handler_data['header']
+            if not handler_data.get('allow_main', False):
+                submission_prefix += '#define main main_%s\n' % str(uuid.uuid4()).replace('-', '')
```

Comment:
```suggestion
                submission_prefix += '#define main main_%s\n' % uuid.uuid4().hex
```
I think these should be equivalent, and grabbing the hex directly makes it more clear what we are trying to do.